### DISCUSSION TITLE
Random sequence sampling from a time series for training fixed size input models

### DISCUSSION THREAD

**Discussion by DAT-FYAYC** [Nixtla/neuralforecast#455] [Ideas]
Does the framework provide random sequence sampling from a time series to use for training for fixed input size models?
Given we have a ts with 100 data points and I want to train a model with a fixed input size (e.g. TFT with encoder length of 6) and a fixed prediction horizon (h=3). When training the model it would make sense to take a random subsequence of size 9 (6+3) within the range of the available 100 data points (for each epoch) to ensure the model is getting trained over the full time period to capture all information within the time series. Is this already implemented when training such models?

**Comment by kdgutier** [comment]
Hi @davidtiefenthaler,

Regarding the usage of time series windows, we have two families of models in the NeuralForecast library:
- ``BaseWindows:`` Generates as many windows of size lag+horizon as the series length's allows, and samples them randomly during the train procedure. Examples of these models are `NBEATS`, `NHITS`, `MLP`, `TFT`.
- ``BaseRecurrent:`` Makes a single pass through the series length with a recurrent encoder and decodes each time step with the forked sequences technique. Examples of these models are `RNN`, `LSTM`, `TCN`.

Here is documentation for `MLP` and `DilatedRNN` that depict the windows and the recurrent strategies:
- https://nixtla.github.io/neuralforecast/models.mlp.html
- https://nixtla.github.io/neuralforecast/models.dilated_rnn.html

**Reply by DAT-FYAYC** [reply]
Thank you for the clarification. That´s what I was thinking about.
It might make sense to add this as an individual section (same level as "models" or as a sub section of "Train/Evaluation") in the documentation since it it very hard to find how training for the different model types is actually done.
Cheers David

**Reply by DAT-FYAYC** [reply]
Or just add this information here: https://nixtla.github.io/neuralforecast/common.base_windows.html
Since it is not mentioned there :)

**Reply by kdgutier** [reply]
Thanks for the suggestion, we need to make a general refactor of the NeuralForecast documentation.
Adding the `BaseWindows` and `BaseRecurrent` classes to the Train/Evaluation index is a good idea.
